journal/src/main/java/io/zeebe/journal/file/MappedJournalSegmentReader.java

```diff
@@ -44,89 +43,53 @@
     reset();
   }
 
-  public JournalRecord getCurrentEntry() {
-    return currentEntry;
-  }
-
-  public JournalRecord getNextEntry() {
-    return nextEntry;
-  }
-
   public boolean hasNext() {
-    // If the next entry is null, check whether a next entry exists.
-    if (nextEntry == null) {
-      readNext(getNextIndex());
-    }
-    return nextEntry != null;
+    // if the next entry exists the version would be non-zero
+    return FrameUtil.hasValidVersion(buffer, buffer.position());
   }
 
   public JournalRecord next() {
     if (!hasNext()) {
       throw new NoSuchElementException();
     }
 
-    // Set the current entry to the next entry.
-    currentEntry = nextEntry;
-
-    // Reset the next entry to null.
-    nextEntry = null;
-
-    // Read the next entry in the segment.
-    readNext(getNextIndex());
-
-    // Return the current entry.
+    final Optional<Integer> version = FrameUtil.readVersion(buffer);
+    if (version.isEmpty()) {
+      return null;
+    }
```

Comment:
If we already checked the validity of the entry then we don't need this check. Maybe the optional no longer makes sense given the new usage

Reply:
Agree. I will remove the optional, and make sure we call `hasVersion` before every `readVersion`.